p/scripts/main.js

```diff
@@ -1068,6 +1078,14 @@ function init_stream(stream) {
 		}
 	};
 
+	if (!navigator.share) {	// https://developer.mozilla.org/en-US/docs/Web/API/Navigator/share
+		document.querySelectorAll('.item.share > button[data-type="web-sharing-api"]').forEach(
+			function (item) {
+				item.style.display = 'none';
```

Comment:
I'd reverse the direction.

Reply:
I do not understand what you mean

Reply:
if navigator.share etc.

Reply:
most common browsers support navigator.share, except Firefox
![grafik](https://user-images.githubusercontent.com/1645099/158267077-b3df505b-501a-47b0-b0a5-39c551cf13fb.png)

Reply:
I'm not sure what you think I said but I don't see how that's relevant to my remark about code architecture. :-)

Reply:
As an aside, my Firefox definitely seems to support it.

Reply:
Still I do not understand your `I'd reverse the direction.`. What would you do? Could you support with an example?

> As an aside, my Firefox definitely seems to support it.
Which Firefox Version on which operating system?

Reply:
98 on Linux.

> Still I do not understand your I'd reverse the direction.. What would you do? Could you support with an example?

Identical in effect, but if navigator.share etc. instead of if !navigator.share etc. That way you have the standard feature detection pattern in line with expectations of future readers.

Reply:
I don't get the point what is wrong to hide the menu item for a minor set of browsers?
Why should the `if` check for the majority (navigator.share)? The menu items are generated in `integration.phtml` It is much more easier to list all sharing item and then hide the 1 item, that does not work, than do not list the item and add it via javascript.

Reply:
I'm still not sure what you think I said; reversing the logic shouldn't make anything any harder or easier.

But now that you mention it, you could simplify it along these lines, for performance reasons presumably still preferably with a class name, but in this sample with an attribute selector:

```css
button[data-type="web-sharing-api"] {display:none}
```

```js
if (navigator.share) {
 document.styleSheets[0].insertRule('button[data-type="web-sharing-api"] {display:initial}', document.styleSheets[0].cssRules.length)
}
```

Or the other way around, but as I said I think that's slightly worse for future ease of reading.
```js
if (!navigator.share) {
 document.styleSheets[0].insertRule('button[data-type="web-sharing-api"] {display:none}', document.styleSheets[0].cssRules.length)
}
```

Reply:
PS Also for the ease of future reading, I actually mean that as something more along these lines in the final result, not as my quick one-liner from the console. ;-)
```js
if (navigator.share) {
  document.styleSheets[0].insertRule(`
    button[data-type="web-sharing-api"] {
      display: initial;
    }`,
    document.styleSheets[0].cssRules.length)
}

Reply:
Thanks. It helped me a lot.

Reply:
Fun fact, dom.webshare.enabled is set to **false** in my Firefox, yet it works. Weird.